File: tiq-e/strings/1/main.go
package main

import (
	"fmt"
	"time"
)

func reverseString(s []byte) {
	half, max := len(s)>>1, len(s)-1
	for i := 0; i < half; i++ {
		s[i], s[max-i] = s[max-i], s[i]
	}
}

func main() {
	arr := []byte{'h', 'e', 'l', 'l', 'o'}

	var start = time.Now()
	reverseString(arr)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", arr)
}

File: tiq-e/strings/1/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let (half, max) = (s.len() >> 1, s.len() - 1);
        for i in 0..half {
            s.swap(i, max - i)
        }
    }
}

fn main() {
    let mut arr = vec!['h', 'e', 'l', 'l', 'o'];

    let start = Instant::now();
    Solution::reverse_string(&mut arr);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", arr);
}
